### PR TITLE
Preferences, Settings, and New Navigation Added

### PR DESCRIPTION
**Pull request by bdesollar**
<!Pull Request Template>

## Description

This pull request introduces several significant updates to the application, focusing on enhancing navigation and user settings management. Key features added include Tab Navigation and Drawer Navigation, which provide a more intuitive and accessible way for users to navigate through the app. Additionally, we've implemented a SignOut feature, allowing users to securely exit their accounts. The Settings and Preferences pages have been introduced, offering users the ability to customize their app experience according to their needs. Alongside these functional updates, we've made several UI updates to improve the overall look and feel of the application. All new pages and features have been accompanied by comprehensive tests to ensure reliability and functionality.

Dependencies required for this change include:
- `react-navigation` for implementing Tab and Drawer navigation.
- `@react-native-community/async-storage` for handling user preferences and settings data.

## Tests

Tests have been added for all new pages and features, ensuring that:
- Navigation flows work as expected.
- The SignOut process correctly clears user data and navigates to the login screen.
- Settings and Preferences pages correctly update user preferences and settings.
- UI updates render correctly across different device sizes and orientations.

## UI Changes

- Introduced Tab Navigation and Drawer Navigation for improved app navigation.
- Added a new SignOut button within the Drawer Navigation for easy access.
- Designed and implemented the Settings and Preferences pages with a user-friendly interface.
- General UI updates to enhance visual appeal and user experience.
-
image
image
image
image

## Issues

This PR addresses the following issues:
#15 
#9 
#6 
#83 
#71 
#67 

## Additional Notes

This update represents a significant overhaul of the app's navigation and settings management, aiming to provide a more intuitive and customizable user experience. Future updates may include further refinements to the UI and additional settings options based on user feedback.

---

### CODE REVIEW

File: mobile/screens/Home.js [hunk 1]
```diff
@@ -1,118 +1,78 @@
...
-        setAlertText('Error updating preferences');
-        setAlertVisible(true);
-        await timeout(1000);
-        setAlertVisible(false);
-      });
-  }
-  function welcome() {
-    return (
-      <Dialog visible={welcomeVisible} onDismiss={() => welcomeToTaskDialog()}>
-        <Dialog.Title>Welcome</Dialog.Title>
```
Comment (saanbe16) [inline mobile/screens/Home.js:61]: @bdesollar did you move this functionality for the dialog box being displayed on login somewhere else? or just completey removed it? We want to keep this functionality

**Comment by saanbe16** [comment]
@bdesollar Could you please make your PR name more descriptive of what you did, so its easy to refer to in the future?

**Comment by saanbe16** [comment]
Also, in the future we should probably add the coverage directories to gitignore, I can't see any benifit to us pushing those to git

File: mobile/screens/Home.js (hunk 1 above)
Comment (bdesollar) [reply mobile/screens/Home.js:61]: I can move it back

**Comment by bdesollar** [comment]
> Also, in the future we should probably add the coverage directories to gitignore, I can't see any benifit to us pushing those to git

@saanbe16 it is but for some reason it committed...

**Comment by saanbe16** [comment]
> > Also, in the future we should probably add the coverage directories to gitignore, I can't see any benifit to us pushing those to git
> 
> @saanbe16 it is but for some reason it committed...

Weird, we can look at that later

**Review by saanbe16 — APPROVED** [review]

**Pull request merged**